atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -792,8 +794,13 @@ private static LockAndTimestampServices getLockAndTimestampServices(
             String userAgent) {
         LockService lockService = new ServiceCreator<>(metricsManager, LockService.class, userAgent)
                 .applyDynamic(timelockServerListConfig);
-        TimelockService timelockService = new ServiceCreator<>(metricsManager, TimelockService.class, userAgent)
-                .applyDynamic(timelockServerListConfig);
+
+        TimelockServerInterface rawTimelockService = new ServiceCreator<>(metricsManager,
+                TimelockServerInterface.class,
+                userAgent).applyDynamic(timelockServerListConfig);
+
+        TimelockService timelockService = LeasingTimelockClient.create(rawTimelockService);
```

Comment:
I will open a separate pr to actually use the leasing client. So we can merge this first, release Timelock, then merge the pr that uses the leasing client and declares a dependency to correct version of Timelock.